core/src/test/java/io/grpc/internal/SpiffeUtil2Test.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2024 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.internal;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+
+import com.google.common.base.Optional;
+import io.grpc.internal.SpiffeUtil2.SpiffeBundle;
+import io.grpc.testing.TlsTesting;
+import io.grpc.util.CertificateUtils;
+import java.io.IOException;
+import java.nio.file.NoSuchFileException;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateParsingException;
+import java.security.cert.X509Certificate;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Test;
+
+public class SpiffeUtil2Test {
+
+  private static final String SPIFFE_PEM_FILE = "spiffe_cert.pem";
+  private static final String SPIFFE_MULTI_VALUES_PEM_FILE = "spiffe_cert_multi.pem";
+  private static final String SERVER_0_PEM_FILE = "server0.pem";
+  private static final String TEST_DIRECTORY_PREFIX = "io/grpc/internal/";
+  private static final String SPIFFE_TRUST_BUNDLE_FILE = "spiffebundle.json";
+  private static final String SPIFFE_TRUST_BUNDLE_MALFORMED = "spiffebundle_malformed.json";
+  private static final String SPIFFE_TRUST_BUNDLE_WRONG_ELEMENTS =
+      "spiffebundle_wrong_elements.json";
+  private static final String SPIFFE_TRUST_BUNDLE_DUPLICATES = "spiffebundle_duplicates.json";
+  private static final String SPIFFE_TRUST_BUNDLE_WITH_WRONG_ROOT =
+      "spiffebundle_wrong_root.json";
+
+
+  private X509Certificate[] spiffeCert;
+  private X509Certificate[] spiffeMultiCert;
+  private X509Certificate[] serverCert0;
+
+
+  @Before
+  public void setUp() throws CertificateException {
+    spiffeCert = CertificateUtils.getX509Certificates(TlsTesting.loadCert(SPIFFE_PEM_FILE));
+    spiffeMultiCert = CertificateUtils.getX509Certificates(TlsTesting
+        .loadCert(SPIFFE_MULTI_VALUES_PEM_FILE));
+    serverCert0 = CertificateUtils.getX509Certificates(TlsTesting.loadCert(SERVER_0_PEM_FILE));
+  }
+
+  @Test
+  public void extractSpiffeIdSuccessTest() throws CertificateParsingException {
+    Optional<SpiffeUtil2.SpiffeId> spiffeId = SpiffeUtil2.extractSpiffeId(spiffeCert);
+    assertEquals("foo.bar.com", spiffeId.get().getTrustDomain());
+    assertEquals("client/workload/1", spiffeId.get().getPath());
```

Comment:
nit: Shouldn't the path be prefixed with "/"?

Reply:
Correct, this is a deficiency of this dummy impl. It's fixed after merging with real parsing library at https://github.com/grpc/grpc-java/pull/11575/files#diff-fdbe5016403200b9fc7adab29732582366f2494986933098d86fddd1081d6a0bR241